kratos/containers/variable_data.cpp

```diff
@@ -22,42 +22,33 @@
 #include "includes/define.h"
 #include "containers/variable_data.h"
 #include "input_output/logger.h"
+#include "includes/fnv_1a_hash.h"
 
 
 namespace Kratos
 {
 
     /// Constructor.
-	VariableData::VariableData(const std::string& NewName, std::size_t NewSize, bool Iscomponent) : mName(NewName), mKey(0), mSize(NewSize), mIsComponent(Iscomponent) {}
+	VariableData::VariableData(const std::string& NewName, std::size_t NewSize, bool Iscomponent, char ComponentIndex) : mName(NewName), mKey(0), mSize(NewSize), mIsComponent(Iscomponent) {
+        mKey = GenerateKey(mName, mSize, Iscomponent, ComponentIndex);
+    }
 
 	/// Copy constructor
     VariableData::VariableData(const VariableData& rOtherVariable)
         : mName(rOtherVariable.mName), mKey(rOtherVariable.mKey), mSize(rOtherVariable.mSize), mIsComponent(rOtherVariable.mIsComponent) {}
 
-	VariableData::KeyType VariableData::GenerateKey(const std::string& Name, std::size_t Size, std::size_t ComponentIndex)
+	VariableData::KeyType VariableData::GenerateKey(const std::string& Name, std::size_t Size, bool IsComponent, char ComponentIndex)
 	{
-		// For generation of 32bit hash key I use the 
-		// Jenkins's one-at-a-time hash taken from wikipedia 
-		// I could have used better hash functions like murmur or lookup3
-		// But finally choose this for simplicity. Pooyan.
-		// https://en.wikipedia.org/wiki/Jenkins_hash_function
-
-		unsigned int hash, i;
-		for(hash = i = 0; i < Name.size(); ++i)
-		{
-			hash += Name[i];
-			hash += (hash << 10);
-			hash ^= (hash >> 6);
-		}
-		hash += (hash << 3);
-		hash ^= (hash >> 11);
-		hash += (hash << 15);
-
-		//if(Size > 127) // to be store in a char
-		//	KRATOS_ERROR << "A variable of Kratos cannot be larger than 127 bytes and variable " << Name << " has sizeof " << Size;
-//		KeyType key = hash;
-//		key << 2; // This is for adding the 
-		return hash;
+        std::uint64_t key = Size;
```

Comment:
why are u using explicitly the uint64_t here and size_t in the function signature?

Reply:
I will change also the return type in future. But for now the idea is to have the overflow (in 32bit compilation) at the end and not in the middle of operations.

BTW, a precise review! Thanks!